dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp

```diff
@@ -468,3 +472,55 @@ void CommandBarFlyoutCommandBar::UpdateTemplateSettings()
         }
     }
 }
+
+void CommandBarFlyoutCommandBar::UpdateShadow()
+{
+    if (PrimaryCommands().Size() > 0)
+    {
+        AddShadow();
+    }
+    else if (PrimaryCommands().Size() == 0)
+    {
+        ClearShadow();
+    }
+}
+
+void CommandBarFlyoutCommandBar::AddShadow()
+{
+    if (SharedHelpers::IsThemeShadowAvailable())
+    {
+#ifdef USE_INSIDER_SDK
+        //Apply Shadow on the Grid named "ContentRoot", this is the first element below
+        //the clip animation of the commandBar. This guarantees that shadow respects the 
+        //animation
+        winrt::IControlProtected thisAsControlProtected = *this;
+        auto grid = GetTemplateChildT<winrt::Grid>(L"ContentRoot", thisAsControlProtected);
+
+        if (grid != nullptr && grid.Shadow() == nullptr)
```

Comment:
Please use try_as to defend against the IUIElement10 API changing on different builds.